### ISSUE TITLE
Installation Instructions Missing Something

### ISSUE THREAD

**Issue by DarioS** [statisticalbiotechnology/diffacto#17]
I am using vitrualenv and pip install. The packages listed installed fine, but it's unclear if that also installs diffacto or not.

```
Installing collected packages: threadpoolctl, joblib, decorator, scikit-learn, pyteomics, networkx
Successfully installed decorator-4.4.2 joblib-1.0.1 networkx-2.5.1 pyteomics-4.4.2 scikit-learn-0.24.1 threadpoolctl-2.1.0
(Python3) /verona/nobackup$ run_diffacto.py
-bash: run_diffacto.py: command not found
```

Meanwhile `pip install cutadapt`, another Python software, works straight away.

```
(Python3) /verona/nobackup$ cutadapt
This is cutadapt 3.4 with Python 3.7.3
Run "cutadapt --help" to see command-line options
```

Is there something important missing from the installation instructions for pip install for diffacto?

**Comment by percolator** [comment]
Hi Dario,

 here is the log I get when I install and run diffacto, can you see if you
get a different response?

(base) ***@***.***:~$ conda activate test_diffacto
(test_diffacto) ***@***.***:~$ conda install python pip
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/lukask/anaconda3/envs/test_diffacto

  added / updated specs:
    - pip
    - python


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    ca-certificates-2021.4.13  |       h06a4308_1         114 KB
    python-3.9.4               |       hdb3f193_0        18.2 MB
    sqlite-3.35.4              |       hdfb4753_0         981 KB
    ------------------------------------------------------------
                                           Total:        19.3 MB

The following NEW packages will be INSTALLED:

  _libgcc_mutex      pkgs/main/linux-64::_libgcc_mutex-0.1-main
  ca-certificates
 pkgs/main/linux-64::ca-certificates-2021.4.13-h06a4308_1
  certifi            pkgs/main/linux-64::certifi-2020.12.5-py39h06a4308_0
  ld_impl_linux-64   pkgs/main/linux-64::ld_impl_linux-64-2.33.1-h53a641e_7
  libffi             pkgs/main/linux-64::libffi-3.3-he6710b0_2
  libgcc-ng          pkgs/main/linux-64::libgcc-ng-9.1.0-hdf63c60_0
  libstdcxx-ng       pkgs/main/linux-64::libstdcxx-ng-9.1.0-hdf63c60_0
  ncurses            pkgs/main/linux-64::ncurses-6.2-he6710b0_1
  openssl            pkgs/main/linux-64::openssl-1.1.1k-h27cfd23_0
  pip                pkgs/main/linux-64::pip-21.0.1-py39h06a4308_0
  python             pkgs/main/linux-64::python-3.9.4-hdb3f193_0
  readline           pkgs/main/linux-64::readline-8.1-h27cfd23_0
  setuptools         pkgs/main/linux-64::setuptools-52.0.0-py39h06a4308_0
  sqlite             pkgs/main/linux-64::sqlite-3.35.4-hdfb4753_0
  tk                 pkgs/main/linux-64::tk-8.6.10-hbc83047_0
  tzdata             pkgs/main/noarch::tzdata-2020f-h52ac0ba_0
  wheel              pkgs/main/noarch::wheel-0.36.2-pyhd3eb1b0_0
  xz                 pkgs/main/linux-64::xz-5.2.5-h7b6447c_0
  zlib               pkgs/main/linux-64::zlib-1.2.11-h7b6447c_3


Proceed ([y]/n)?


Downloading and Extracting Packages
sqlite-3.35.4        | 981 KB    |
###########################################################################
| 100%
ca-certificates-2021 | 114 KB    |
###########################################################################
| 100%
python-3.9.4         | 18.2 MB   |
###########################################################################
| 100%
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
(test_diffacto) ***@***.***:~$ pip install diffacto
Collecting diffacto
  Downloading diffacto-1.0.5-py36-none-any.whl (14 kB)
Collecting numpy>=1.10
  Downloading numpy-1.20.2-cp39-cp39-manylinux2010_x86_64.whl (15.4 MB)
     |████████████████████████████████| 15.4 MB 345 kB/s
Collecting pyteomics>=3.3
  Downloading pyteomics-4.4.2-py2.py3-none-any.whl (196 kB)
     |████████████████████████████████| 196 kB 573 kB/s
Collecting scikit-learn>=0.17
  Downloading scikit_learn-0.24.1-cp39-cp39-manylinux2010_x86_64.whl (23.8
MB)
     |████████████████████████████████| 23.8 MB 643 kB/s
Collecting scipy>=0.17
  Downloading scipy-1.6.2-cp39-cp39-manylinux1_x86_64.whl (27.3 MB)
     |████████████████████████████████| 27.3 MB 524 kB/s
Collecting networkx>=1.10
  Downloading networkx-2.5.1-py3-none-any.whl (1.6 MB)
     |████████████████████████████████| 1.6 MB 197 kB/s
Collecting pandas>=0.18
  Downloading pandas-1.2.4-cp39-cp39-manylinux1_x86_64.whl (9.7 MB)
     |████████████████████████████████| 9.7 MB 67 kB/s
Collecting decorator<5,>=4.3
  Downloading decorator-4.4.2-py2.py3-none-any.whl (9.2 kB)
Collecting python-dateutil>=2.7.3
  Using cached python_dateutil-2.8.1-py2.py3-none-any.whl (227 kB)
Collecting pytz>=2017.3
  Downloading pytz-2021.1-py2.py3-none-any.whl (510 kB)
     |████████████████████████████████| 510 kB 687 kB/s
Collecting six>=1.5
  Using cached six-1.15.0-py2.py3-none-any.whl (10 kB)
Collecting threadpoolctl>=2.0.0
  Downloading threadpoolctl-2.1.0-py3-none-any.whl (12 kB)
Collecting joblib>=0.11
  Downloading joblib-1.0.1-py3-none-any.whl (303 kB)
     |████████████████████████████████| 303 kB 574 kB/s
Installing collected packages: six, numpy, threadpoolctl, scipy, pytz,
python-dateutil, joblib, decorator, scikit-learn, pyteomics, pandas,
networkx, diffacto
Successfully installed decorator-4.4.2 diffacto-1.0.5 joblib-1.0.1
networkx-2.5.1 numpy-1.20.2 pandas-1.2.4 pyteomics-4.4.2
python-dateutil-2.8.1 pytz-2021.1 scikit-learn-0.24.1 scipy-1.6.2
six-1.15.0 threadpoolctl-2.1.0
(test_diffacto) ***@***.***:~$ diffacto -h
usage: diffacto [-h] -i I [-db [DB]] [-samples [SAMPLES]] [-log2 LOG2]
[-normalize {average,median,GMM,None}]
                [-farms_mu FARMS_MU] [-farms_alpha FARMS_ALPHA] [-reference
REFERENCE] [-min_samples MIN_SAMPLES]
                [-use_unique USE_UNIQUE] [-impute_threshold
IMPUTE_THRESHOLD] [-cutoff_weight CUTOFF_WEIGHT]
                [-fast FAST] [-out OUT] [-mc_out MC_OUT]

optional arguments:
  -h, --help            show this help message and exit
  -i I                  Peptides abundances in CSV format. The first row
should contain names for all samples. The
                        first column should contain unique peptide
sequences. Missing values should be empty instead
                        of zeros. (default: None)
  -db [DB]              Protein database in FASTA format. If None, the
peptide file must have protein ID(s) in the
                        second column. (default: None)
  -samples [SAMPLES]    File of the sample list. One run and its sample
group per line, separated by tab. If None,
                        read from peptide file headings, then each run will
be summarized as a group. (default:
                        None)
  -log2 LOG2            Input abundances are in log scale (True) or linear
scale (False) (default: False)
  -normalize {average,median,GMM,None}
                        Method for sample-wise normalization. (default:
None)
  -farms_mu FARMS_MU    Hyperparameter mu (default: 0.1)
  -farms_alpha FARMS_ALPHA
                        Hyperparameter weight of prior probability
(default: 0.1)
  -reference REFERENCE  Names of reference sample groups (separated by
semicolon) (default: average)
  -min_samples MIN_SAMPLES
                        Minimum number of samples peptides needed to be
quantified in (default: 1)
  -use_unique USE_UNIQUE
                        Use unique peptides only (default: False)
  -impute_threshold IMPUTE_THRESHOLD
                        Minimum fraction of missing values in the group.
Impute missing values if missing fraction
                        is larger than the threshold. (default: 0.99)
  -cutoff_weight CUTOFF_WEIGHT
                        Peptides weighted lower than the cutoff will be
excluded (default: 0.5)
  -fast FAST            Allow early termination in EM calculation when
noise is sufficiently small. (default: False)
  -out OUT              Path to output file (writing in TSV format).
(default: <_io.TextIOWrapper name='<stdout>'
                        mode='w' encoding='utf-8'>)
  -mc_out MC_OUT        Path to MCFDR output (writing in TSV format).
(default: None)
(test_diffacto) ***@***.***:~$

**Comment by DarioS** [comment]
Oh, the server we use doesn't have conda installed, so I use pip directly. The intructions in the section titled Installation Via pip seem to only download the dependencies. I realised that I need to clone the Diffacto respository and run `python setup.py install`. Could there be some clarification about how users without conda install Diffacto? I assumed that it was in PyPi and I could use pip,

**Comment by percolator** [comment]
This is strange. As I wanted to demonstrate with my previous email, when
executed in a clean environment, pip installs all dependencies
automatically, and "diffacto -h" gives a usage statement. Would you be able
to fill in more details about your environment?

--Lukas

**Comment by markmipt** [comment]
Just would like to add my feedback which could help you:

I have linux Arch, don't have conda installed and use pip directly (pip install diffacto). And I can run diffacto command after diffacto was installed into a fresh virtual environment. I have checked it for both Python3.7 (which Dario has) and the latest Python3.9

However, PyPi version seems outdated and that is why I have to use github version for real installation and usage:
pip install https://github.com/statisticalbiotechnology/diffacto/archive/master.zip

Also I suppose that there is a misunderstanding with the running command: Dario is trying to use “run_diffacto.py” command to run Diffacto, Lukas is using “diffacto” and the Readme page tells us to use “diffacto.py”.

Regards,
Mark

**Comment by DarioS** [comment]
Thanks for changing the instructions. It now works as I expected.

```
virtualenv -p python3 test
source test/bin/activate
pip install diffacto
```

**Comment by percolator** [comment]
Thanks both!
